packages/flutter_test/pubspec.yaml

```diff
@@ -16,6 +16,7 @@ dependencies:
   # 'test' package, which change between versions, so when upgrading
   # this, make sure the tests are still running correctly.
   test_api: 0.6.1
+  test_core: 0.5.9
```

Comment:
We somehow need to ensure that the frontend server knows to include `package:test_core` and `package:ffi` when compiling `root_test_isolate_spawner.dart` and `child_test_isolate_spawner.dart`. Is it ok if I include them as dependencies of `package:flutter_test`?

Reply:
This would need further discussion. See this error from `Linux analyze`:

```
The transitive closure of package dependencies contains non-allowlisted packages:
  test_core, yaml, source_maps, source_map_stack_trace, pool, package_config, io, glob, frontend_server_client, coverage, logging, args, analyzer, watcher, pub_semver, crypto, typed_data, convert, _fe_analyzer_shared, ffi
We strongly desire to keep the number of dependencies to a minimum and
therefore would much prefer not to add new dependencies.
See dev/bots/allowlist.dart for instructions on how to update the package
allowlist if you nonetheless believe this is a necessary addition.
```

This change essentially lead to all of the listed packages being exactly pinned for all user projects running tests.

Reply:
Can't we make these dependencies of the generated file instead?

Reply:
I believe that to compile the generated files successfully, the frontend server needs to be passed a `package_config.json` that contains the union of `package:test_core`, `package:ffi`, and all the dependencies of the tests. I'll look into generating such a `package_config.json`. Let me know if you think I'm missing something, or have suggestions.

Reply:
I've implemented `package_config.json` generation based on our discussion on Thursday. I've added `package:ffi` as a `dev_dependency` of `package:flutter_tools`, because we would have to copy over a lot of code otherwise (we use `toNativeUtf8` which requires `MallocAllocator`).